### ISSUE TITLE
ifcopenshell.geom.settings() object has no attribute 'USE_PYTHON_OPENCASCADE'

### ISSUE THREAD

**Issue by J-YingHuang** [IfcOpenShell/IfcOpenShell#758]
Hi, I tried to use ifcopenshell to write the IFC simple viewer.
But I got some error....

Error Message:
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2019.3.1\plugins\python-ce\helpers\pydev\_pydev_bundle\pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2019.3.1\plugins\python-ce\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "E:/JingYing/2019_Sino_RIMSystem/Viewer/PythonOCC/MainViewer.py", line 4, in <module>
    settings.set(settings.USE_PYTHON_OPENCASCADE, True)
  File "C:\Users\User\Anaconda3\envs\pythonocc_OpenShell\lib\site-packages\ifcopenshell\ifcopenshell_wrapper.py", line 104, in <lambda>
    __getattr__ = lambda self, name: _swig_getattr(self, settings, name)
  File "C:\Users\User\Anaconda3\envs\pythonocc_OpenShell\lib\site-packages\ifcopenshell\ifcopenshell_wrapper.py", line 80, in _swig_getattr
    raise AttributeError("'%s' object has no attribute '%s'" % (class_type.__name__, name))
AttributeError: 'settings' object has no attribute 'USE_PYTHON_OPENCASCADE'

How can I resolve this error?

**Comment by aothms** [comment]
You need to have PythonOCC installed. What does `import OCC.BRep` say?

**Comment by moonlikestone** [comment]
Hi, before I wrote those code, I has installed PythonOCC and ifcopenshell in the same time.
But the problem still occurred...   

This is the list of packages in my environment:
1. boost-cpp                 1.70.0           
2. certifi                   2019.11.28            
3. fontconfig                2.13.1           
4. freeimageplus             3.18.0              
5. freetype                  2.10.0              
6. icu                       64.2                
7. ifcopenshell              v0.6.0b0         
8. libiconv                  1.15              
9. libpng                    1.6.37              
10. libxml2                   2.9.10              
11. occt                      7.4.0                
12. opencascade               7.4.0               
13. pip                       19.3.1                   
14. pyqt5                     5.14.1                  
15. pyqt5-sip                 12.7.0                  
16. python                    3.6.7             
17. pythonocc-core            7.4.0rc1         
18. setuptools                44.0.0                   
19. six                       1.13.0              
20. tbb                       2019.9           
21. vc                        14.1             
22. vs2015_runtime            14.16.27012         
23. wheel                     0.33.6                  
24. wincertstore              0.2                   
25. zlib                      1.2.11

**Comment by khyatiip** [comment]
I have the same issue.

I'm using Ananconda 3 (Spyder) to run my code and to install pythonocc and ifcopenshell and it gives me the same error.

settings = ifcopenshell.geom.settings()
When i print(dir(settings) USE_PYTHON_OPENCASCADE is not in the list.

I have attached the screenshot of the code and the packages installed using conda installer.

<img width="960" alt="Screenshot 2020-01-08 03 18 26" src="https://user-images.githubusercontent.com/44809431/71978675-38b3c500-321c-11ea-9c80-1827b92b8115.png">
<img width="960" alt="Screenshot 2020-01-08 03 18 40" src="https://user-images.githubusercontent.com/44809431/71978676-38b3c500-321c-11ea-92f3-f79aacf67aa9.png">
<img width="960" alt="Screenshot 2020-01-08 13 35 46" src="https://user-images.githubusercontent.com/44809431/71978677-38b3c500-321c-11ea-8ebb-4434274a684d.png">

**Comment by aothms** [comment]
@moonlikestone 

> pythonocc-core 7.4.0rc1

Was not supported yet at the time of the latest builds. You can apply https://github.com/IfcOpenShell/IfcOpenShell/pull/737 this to your IfcOpenShell module folder locally or use an older version of pythonocc

**Comment by khyatiip** [comment]
> @moonlikestone
> 
> > pythonocc-core 7.4.0rc1
> 
> Was not supported yet at the time of the latest builds. You can apply #737 this to your IfcOpenShell module folder locally or use an older version of pythonocc

@moonlikestone 

I did not use this but i got it working. In ifcopenshell/geom/main.py and ifcopenshell/geom/occ_utlis.py wherever `OCC.something` is being used, change it to `OCC.Core.something`.

This may not be the correct/ideal solution but it is working for me as of now.

**Comment by rurubino1** [comment]
> @moonlikestone
> 
> > pythonocc-core 7.4.0rc1
> 
> Was not supported yet at the time of the latest builds. You can apply #737 this to your IfcOpenShell module folder locally or use an older version of pythonocc

Hi,
I tried this but I have a new error, apparently in Filter01.py:
[error2.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/4059731/error2.txt)

Any suggestion?
PS: see my environment setup also attached (it looks the same setup as @khyatiip to me).
[bim2bem_env.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/4059762/bim2bem_env.txt)

**Comment by mieszk** [comment]
@rurubino1 
According to the error message:
```
File "C:\Users\rurubino\.conda\envs\bim2bem36\lib\site-packages\ifcopenshell\geom\main.py", line 72, in <module>
    for ty in (ifcopenshell_wrapper.iterator_single_precision, ifcopenshell_wrapper.iterator_double_precision):
```
It looks like there is a line in the `main.py` code that tries to reach non-existent variable `iterator_single_precision`. Which is kind of weird, because I looked at the code under `v0.6.0b0` release and there is no sign of the `iterator_single_precision`, only the double-precision one. It was removed in `v0.6.0a1` already.

Try to update package with `conda update ifcopenshell` or replace the loop with the new code:
```python
assert ifcopenshell_wrapper.iterator_double_precision.mantissa_size() == sys.float_info.mant_dig
_iterator = ifcopenshell_wrapper.iterator_double_precision
```

**Comment by rurubino1** [comment]
@mieszk 
Thanks for your suggestion.
Your comment on main.py seems strange: in https://github.com/mieszk/IfcOpenShell/commit/81f76eddf952e476360e0dcbd55e34b1d1135ad2#diff-1d9714892cfa45e4254c64b33c701958 I still see the that loop referring to `iterator_single_precision`
I tried to update the package, but it also updated pythonocc-core to 0.18.2 so I tried to revert but I guess something went wrong and now the environment seems broken (dependency check does not find PyQt5 although it's there and can't even update it).
So, I created a new environment from scratch with:
conda create --name bim2bem2020 python=3.6
conda activate bim2bem2020
conda install mako
pip install -U jinja2 --proxy 159.82.13.140
pip install -U lxml --proxy 159.82.13.140
pip install -U PyQt5 --proxy 159.82.13.140
conda install -c dlr-sc opencascade
conda install pythonocc-core
conda install ifcopenshell

Then applied  #736 as suggested by @aothms .
Now I get a different error:
INFO: Starting CoTeTo.Controller from file C:\Users\rurubino\Documents\_RES House Modeling\BIM2BEM\CoTeTo-master\CoTeTo\Controller.py, version 1.0
INFO: IFCFile - loading C:/Users/rurubino/Documents/_RES House Modeling/BIM2BEM/CoTeTo-master/Generators/IFC_MultiZoneBuildings_Modelica/OneZone.ifc
INFO: determine building model information for file: C:/Users/rurubino/Documents/_RES House Modeling/BIM2BEM/CoTeTo-master/Generators/IFC_MultiZoneBuildings_Modelica/OneZone.ifc
CRITICAL: An unhandled exception occured
****************************************

Traceback (most recent call last):
  File "C:\Users\rurubino\Documents\_RES House Modeling\BIM2BEM\CoTeTo-master\CoTeTo\GUI.py", line 269, in executeGenerator
    x = self.activeGenerator.execute(uriList, outputBase)
  File "C:\Users\rurubino\Documents\_RES House Modeling\BIM2BEM\CoTeTo-master\CoTeTo\Generator.py", line 262, in execute
    self.executeFilter()
  File "C:\Users\rurubino\Documents\_RES House Modeling\BIM2BEM\CoTeTo-master\CoTeTo\Generator.py", line 187, in executeFilter
    function(self.data, self.controller.systemCfg, self.cfg, self.logger)
  File "C:\Users\rurubino\Documents\_RES House Modeling\BIM2BEM\CoTeTo-master\Generators\IFC_MultiZoneBuildings_Modelica\Filters\Filter01.py", line 497, in filter01
    dataModel = mapIFCtoBuildingDataModel(fileData['IfcData'], fileName)
  File "C:\Users\rurubino\Documents\_RES House Modeling\BIM2BEM\CoTeTo-master\Generators\IFC_MultiZoneBuildings_Modelica\Filters\Filter01.py", line 61, in mapIFCtoBuildingDataModel
    mesh = DataClasses.Mesh(toposhape)
  File "C:\Users\rurubino\Documents\_RES House Modeling\BIM2BEM\CoTeTo-master\Generators\IFC_MultiZoneBuildings_Modelica\Filters\IfcLib\DataClasses.py", line 106, in __init__
    facing = (bt.Triangulation(face, location)).GetObject()
AttributeError: 'Poly_Triangulation' object has no attribute 'GetObject'
****************************************

PS: Should probably move this discussion to https://github.com/UdK-VPT/BIM2Modelica/issues/3# ?

**Comment by mieszk** [comment]
@rurubino1 
> Your comment on main.py seems strange: in [mieszk@81f76ed#diff-1d9714892cfa45e4254c64b33c701958](https://github.com/mieszk/IfcOpenShell/commit/81f76eddf952e476360e0dcbd55e34b1d1135ad2#diff-1d9714892cfa45e4254c64b33c701958) I still see the that loop referring to `iterator_single_precision`

That's right, but that was on master, you should check out the tag matching your `ifcopenshell` version on `conda` (or just branch 0.6.0) and use the code from there.

**Comment by Moult** [comment]
I think this can be closed, since v0.7.0 is now out and there are now up to date conda builds.
